libbpf-rs/tests/test_xdp.rs

```diff
@@ -41,6 +46,10 @@ fn test_sudo_xdp() {
         .query(LO_IFINDEX, XdpFlags::UPDATE_IF_NOEXIST)
         .is_ok());
 
+    assert!(xdp_prog1
+        .replace(LO_IFINDEX, fd.as_raw_fd(), XdpFlags::REPLACE)
```

Comment:
It doesn't really make sense to have a user require specifying the flag `REPLACE` on a replace operation, does it? I presume it would fail or not do the right thing without that?

Also, can you please add a proper function test and not just a mere check for a return code? This test would pass if all the function did was return `Ok(())`.

Reply:
> It doesn't really make sense to have a user require specifying the flag REPLACE on a replace operation, does it? I presume it would fail or not do the right thing without that?

Got it. Should've noticed it. Removed the `flag` arg now.

> Also, can you please add a proper function test and not just a mere check for a return code? This test would pass if all the function did was return Ok(()).

Added the test to check for the old and new prog id. This way we can make sure that the replace had happened.